src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2937,6 +2937,25 @@ uct_ib_mlx5_devx_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
     return UCS_OK;
 }
 
+static UCS_F_ALWAYS_INLINE uct_ib_md_packed_mkey_t
+uct_ib_mlx5_devx_get_packed_mkey(const void *mkey_buffer, uint8_t mkey_size)
+{
+    uct_ib_md_packed_mkey_t *tmp = (uct_ib_md_packed_mkey_t *)mkey_buffer;
+    uct_ib_md_packed_mkey_t packed_mkey;
+
+    packed_mkey.lkey    = tmp->lkey;
+    packed_mkey.vhca_id = tmp->vhca_id;
+    packed_mkey.md_uuid = 0;
+
+    /* Backward compatibility: extract MD UUID from mkey buffer only if it's
+     * present there */
+    if (mkey_size > ucs_offsetof(uct_ib_md_packed_mkey_t, md_uuid)) {
+        packed_mkey.md_uuid = tmp->md_uuid;
+    }
+
+    return packed_mkey;
```

Comment:
IMO we should use "serialize" functions, since the uuid field may not be accessible and it's too dangerous to assign it to "tmp" pointer where last field may not be accessible